PyInstaller/hooks/hook-sklearn.py

```diff
@@ -0,0 +1,4 @@
+from PyInstaller.utils.hooks import collect_data_files
+
+datas = collect_data_files('sklearn')
+hiddenimports = ['sklearn.neighbors.typedefs']
```

Comment:
Is this actually a hidden import of `sklearn`, or from`sklearn.neighbors`? If the later this should go ibnto a hook for `sklearn.neighbors`, since if for any reason, `sklearn.neighbors` is not used/needed, `typedefs` is neither.

Reply:
I just ran into this issue. I use `from sklearn.linear_model import LinearRegression` which causes a ton of imports.

Looks like my import triggers linear_model > linear_model.randomized_l1 > feature_selection > feature_selection.mutual_info_ > neighbors > neighbors.ball_tree > neighbors.dist_metrics > hidden typedefs import (because dist_metrics is just a stub for a pyd file).

Seeing this tangled mess makes me want to drop sklearn for my project because I'm just doing a multiple regression for which statsmodels, numpy, or scipy would be a much lighter dependency.

Reply:
To be fair, I believe sklearn is much lighter than numpy or scipy in filesize.

Reply:
Sure, I just already have them as dependencies. I always figured scikit learn was like pandas - one of the many 2nd tier pydata packages that builds on scipy and numpy.